lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -772,16 +795,19 @@ class TypeRefinementContextBuilder : private ASTWalker {
       FalseRefinement = FalseFlow;
     }
 
-    if (NestedCount == 0)
-      return std::make_pair(None, FalseRefinement);
-
     TypeRefinementContext *NestedTRC = getCurrentTRC();
     while (NestedCount-- > 0)
       ContextStack.pop_back();
 
     assert(getCurrentTRC() == StartingTRC);
 
-    return std::make_pair(NestedTRC->getAvailabilityInfo(), FalseRefinement);
+
+    if (!isUnavailability.hasValue() || !isUnavailability.getValue()) {
+      return std::make_pair(NestedTRC->getAvailabilityInfo(), FalseRefinement);
+    } else {
+      // If this is an unavailability check, invert the result.
+      return std::make_pair(FalseRefinement, NestedTRC->getAvailabilityInfo());
+    }
```

Comment:
```suggestion
    assert(isUnavailability.hasValue());
    if (isUnavailability.getValue()) {
      return std::make_pair(FalseRefinement, NestedTRC->getAvailabilityInfo());
    }
    return std::make_pair(NestedTRC->getAvailabilityInfo(), FalseRefinement);
```

(I would try to avoid the double `!` if possible.)

Reply:
I inverted the position for them to become positive, but I think I can't add the assertion because this method can be called when there are no conditions as well (so `isUnavailability` would be None)